icu4c/source/common/locid.cpp

```diff
@@ -1340,7 +1340,25 @@ Locale::getUnicodeKeywordValue(StringPiece keywordName,
 void
 Locale::setKeywordValue(const char* keywordName, const char* keywordValue, UErrorCode &status)
 {
-    uloc_setKeywordValue(keywordName, keywordValue, fullName, ULOC_FULLNAME_CAPACITY, &status);
+    int32_t bufferLength = uprv_max((int32_t)(uprv_strlen(fullName) + 1), ULOC_FULLNAME_CAPACITY);
+    int32_t newLength = uloc_setKeywordValue(keywordName, keywordValue, fullName,
+                                         bufferLength, &status) + 1;
+    /* Handle the case the current buffer is not enough to hold the new id */
+    if (status == U_BUFFER_OVERFLOW_ERROR || newLength > bufferLength) {
+        char* newFullName = (char *)uprv_malloc(sizeof(char)*(newLength));
```

Comment:
I don't think that `sizeof(char)` adds any value here, it's by definition 1 and I'm quite certain that everyone who ever reads this code knows that.

Reply:
again, see existing code in  Locale::init